### PR TITLE
refact: make offload bucket auto creation configurable

### PR DESCRIPTION
**Pull request by moogacs**
### What's being changed:

this PR 
- adds `OFFLOAD_S3_BUCKET_AUTO_CREATE` to make the offload bucket auto create configurable, so by default now the bucket won't be created and user has to pass `OFFLOAD_S3_BUCKET_AUTO_CREATE=true` to make sure it's auto created.

- adds tests to confirm change

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: modules/offload-s3/module.go [hunk 1]
```diff
@@ -167,45 +168,54 @@ func (m *Module) Init(ctx context.Context,
...
-	if bucket != "" {
+	if bucket := os.Getenv(s3Bucket); bucket != "" {
 		m.Bucket = bucket
 	}
 
-	endpoint := os.Getenv(s3Endpoint)
-	if endpoint != "" {
+	var autoCreateBucket bool
+	if autoCreateBucketStr := os.Getenv(s3BucketAutoCreate); autoCreateBucketStr != "" {
+		v, err := strconv.ParseBool(autoCreateBucketStr)
```
Comment (antas-marcin) [inline modules/offload-s3/module.go:177]: you can use `entcfg` package for this:
```
entcfg "github.com/weaviate/weaviate/entities/config"
```
there is a convenient method `entcfg.Enabled(os.Getenv(s3BucketAutoCreate))`

File: test/docker/compose.go [hunk 2]
```diff
@@ -737,3 +754,25 @@ func copySettings(s map[string]string) map[string]string {
 	}
 	return copy
 }
+
+func createBucket(ctx context.Context, endpoint, region, bucketName string) error {
```
[the remark this answers is not in the document]
Comment (moogacs) [reply test/docker/compose.go:758]: for the offload case the problem is we need to create the bucket in minio before weaviate starts and the current `Start()` func start all in the same call. 

about the backup part i agree we need to combine both, wdyt if we do it as a refactor later on to not have big diff and do the release

**Review by antas-marcin — APPROVED** [review]

File: modules/offload-s3/module.go [hunk 3]
```diff
@@ -31,15 +31,17 @@ import (
 	"github.com/weaviate/weaviate/entities/modulecapabilities"
 	"github.com/weaviate/weaviate/entities/moduletools"
 	"github.com/weaviate/weaviate/usecases/config"
+	"github.com/weaviate/weaviate/usecases/configbase"
```
Comment (antas-marcin) [inline modules/offload-s3/module.go:34]: after merge of my PR you need to change this to:
```
entcfg "github.com/weaviate/weaviate/entities/config"
```

**Review by antas-marcin — DISMISSED** [review]

File: test/docker/compose.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (moogacs) [reply test/docker/compose.go:759]: will open a PR for that

**Review by antas-marcin — APPROVED** [review]

**Pull request merged**